std/experimental/allocator/common.d

```diff
@@ -391,6 +392,7 @@ bool alignedReallocate(Allocator)(ref Allocator alloc,
         if (b.length == s) return true;
```

Comment:
This should also check that `b` has the desired `alignment`.
Could you please add this test?

Thanks,
Edi